src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DescribeLayerTest.java

```diff
@@ -74,5 +74,74 @@ public void testWorkspaceQualified() throws Exception {
         Element e = (Element) dom.getElementsByTagName("LayerDescription").item(0);
         String attribute = e.getAttribute("owsURL");
         assertTrue(attribute.contains("sf/wfs"));
+
+        attribute = e.getAttribute("owsType");
+        assertEquals("WFS", attribute);
+
+        e = (Element) dom.getElementsByTagName("Query").item(0);
+        String typeName = e.getAttribute("typeName");
+        assertEquals("unexpected fully qualified typename", "sf:PrimitiveGeoFeature", typeName);
+    }
+
+    @Test
+    public void testWorkspaceNonQualifiedUrl() throws Exception {
+        Document dom =
+                getAsDOM(
+                        "wms?service=wms&version=1.1.1&request=DescribeLayer"
+                                + "&layers=PrimitiveGeoFeature",
+                        true);
+        // print(dom);
+        assertEquals("WMS_DescribeLayerResponse", dom.getDocumentElement().getNodeName());
+
+        Element e = (Element) dom.getElementsByTagName("LayerDescription").item(0);
+        String attribute = e.getAttribute("owsURL");
+        assertTrue(attribute.contains("wfs"));
+
+        attribute = e.getAttribute("owsType");
+        assertEquals("WFS", attribute);
+
+        e = (Element) dom.getElementsByTagName("Query").item(0);
+        String typeName = e.getAttribute("typeName");
+        assertEquals("unexpected fully qualified typename", "sf:PrimitiveGeoFeature", typeName);
+    }
+
+    @Test
+    public void testImageWMSLayer() throws Exception {
+        Document dom =
+                getAsDOM(
+                        "wms?service=wms&version=1.1.1&request=DescribeLayer&layers=wcs:World",
+                        true);
+        // print(dom);
+        assertEquals("WMS_DescribeLayerResponse", dom.getDocumentElement().getNodeName());
+
+        Element e = (Element) dom.getElementsByTagName("LayerDescription").item(0);
+        String attribute = e.getAttribute("owsURL");
+        assertTrue(attribute.contains("wcs"));
+        attribute = e.getAttribute("owsType");
+        assertEquals("WCS", attribute);
+
+        e = (Element) dom.getElementsByTagName("Query").item(0);
+        String typeName = e.getAttribute("typeName");
+        assertEquals("unexpected fully qualified typename", "wcs:World", typeName);
```

Comment:
Hum... this seems correct, but at the same time, it's not completely... the reason is that the layer is not named in the same way in the various WCS versions. In WCS 1.0 and 1.1 I believe it's ``wcs:World``, but in the WCS 2.0 column are not allowed in coverage names, and the id would be ``wcs__World`` instead.

I don't think it's solvable to be honest, the DTD for DescribeLayer does not allow to specify a version. Just noting the mismatch.

Oh, the error message seems incorrect, shouldn't it have been "Unexpected un-qualified typename"?).

Reply:
yes... and it could also be an empty string; I'll just use "unexpected typename" instead